Documentation/BasicOperators.md

```diff
@@ -46,41 +46,392 @@ code-style will be used.
 ## Performing side effects with signals
 
 ### Observation
+
+Signals can be observed with the `observe` function. It takes a `Sink` as argument to which any future events are sent. 
+
+```Swift
+signal.observe(Signal.Observer { event in
+    switch event {
+    case let .Next(next):
+        println("Next: \(next)")
+    case let .Error(error):
+        println("Error: \(error)")
+    case .Completed:
+        println("Completed")
+    case .Interrupted:
+        println("Interrupted")
+    }
+})
+```
+
+Alternatively, callbacks for the `Next`, `Error`, `Completed` and `Interrupted` events can be provided which will be called when a corresponding event occurs.
+
+```Swift
+signal.observe(next: { next in
+        println("Next: \(next)")
+    }, error: { error in
+        println("Error: \(error)")
+    }, completed: {
+        println("Completed")
+    }, interrupted: {
+        println("Interrupted")
+    }
+)
+```
+
+`observe` is also available as operator that can be used with [|>](#pipe)
+
 ### Injecting effects
 
+Side effects can be injected on a `SignalProducer` with the `on` operator without actually subscribing to it. 
+
+```Swift
+producer
```

Comment:
I've updated the example and added a note about starting the resulting producer.